packages/tinacms/src/client/index.ts

```diff
@@ -179,12 +182,63 @@ mutation addPendingDocumentMutation(
     return this.schema
   }
 
+  /**
+   *
+   * Returns a version of the query with fragments inlined. Eg.
+   * ```graphql
+   * {
+   *   getPostDocument(relativePath: "") {
+   *     data {
+   *       ...PostFragment
+   *     }
+   *   }
+   * }
+   *
+   * fragment PostFragment on Post {
+   *   title
+   * }
+   * ```
+   * Turns into
+   * ```graphql
+   * {
+   *   getPostDocument(relativePath: "") {
+   *     data {
+   *       title
+   *     }
+   *   }
+   * }
+   */
+  getOptimizedQuery = async (documentNode: DocumentNode) => {
```

Comment:
Is there any reason why we don't call `optimizeDocuments` here? is it because we want to keep graphql dependencies on the backend?

Reply:
The library is Node-only, to we need to get it from the server

Reply:
Makes sense!